app/src/main/java/com/github/se/orator/ui/friends/Leaderboard.kt

```diff
@@ -207,3 +239,77 @@ fun LeaderboardItem(rank: Int, profile: UserProfile) {
         }
       }
 }
+
+/**
+ * Composable function that displays a dropdown menu for selecting different rank metrics. The
+ * selected metric is shown and can be changed by the user.
+ */
+@Composable
+fun RankMetricSelector() {
```

Comment:
This function is an excellent addition to the codebase, allowing us to rank profiles dynamically based on various metrics—thank you for implementing this feature! I noticed that the screen dynamically adjusts to rank profiles according to the selected metric, which is great.

However, I observed that the leaderboard item always displays the improvement metric, regardless of the selected ranking metric. Perhaps this could be adjusted to ensure the displayed metric aligns with the selected ranking metric for consistency. What do you think about addressing this? I'd be happy to help if needed

Reply:
You are right. Thank you for noticing it. I will fix this.